### ISSUE TITLE
Useful Tooling

### ISSUE THREAD

**Issue by AlexDawsonUK** [w3c/sustainableweb-wsg#11]
**Issue:** In the CG, we discussed a multitude of ideas for useful tooling that could be created to assist with the adoption and implementation of the WSG. This could help us with measurability, and general ease of compliance, or it could help companies and individuals make the move into becoming more sustainable through using automated tooling (without having to directly reference our work).

**Considerations:** Tooling takes time to make and maintain. Our charter is explicitly about creating the guidelines, not about required tooling (though tooling can be helpful in allowing us to achieve certain goals). Below is a list of ideas that individuals came up with in the CG.

- Google Lighthouse Plugin (Chairs had a meeting with Google staff regarding the potential for this).
- Passport Scheme (See CG issue 93 for a general idea).
- Sustainability Advisor (Choose your own adventure style do-this-then-that).

**Source:**  [CG 54](https://github.com/w3c/sustyweb/issues/54), [CG 93](https://github.com/w3c/sustyweb/issues/93).

**Comment by mgifford** [comment]
I don't think we want to be building tooling. What kinds of tools are there and what kind of information is it that they can collect? What can we learn from the tools and data which can be gathered in the accessibility world?   
  
GoogleLighthouse data is comparable (I think) to Sitespeed.io (which [already includes CO2.js](https://www.thegreenwebfoundation.org/news/sitespeed-io-using-and-contributing-to-co2-js/)).   
  
Just outlining the data that we want that a tool has demonstrated it can collect would be useful.

**Comment by mrchrisadams** [comment]
Hi folks, I very much agree with @mgifford  hre.

While sitespeed.io does include CO2.js with the Sustainable Web Design model, I learned earlier this week that there's been some [more progress on figuring out how to read values _directly_ from a browser in Sitespeed, and automating the process](https://github.com/sitespeedio/browsertime/pull/2220).

It currently works with Firefox, and this is largely because to my knowledge, the Firefox Profiler exposes information that is not currently exposed by Chromium/Blink based browsers, nor ones built on Webkit.

Anyway, I raise this because:

1. I think it would give us an idea about what is possible if we wanted to rely on more than modelled data.
2. It might help with the creation of some convention about what if reasonable to expect from browser vendors.

There's a [nice recent article piece by my colleague Fershad about the grid aware work he's been working](https://fershad.com/writing/making-this-website-grid-aware/#testing-energy-usage-using-firefox-profiler) on with links to Firefox Profiler profile runs. He's been using this to track client-side power usage changes from some new changes he's made to his site. I've linked directly to the part of the post in that link.

**Comment by ldevernay** [comment]
As of today, browsers are making regular breakthough but still suffer from big limitations. Even the Firefox Profiler is still somewhat... inaccurate (even if it gives us insights on the possibilities). 
This topic is to keep in mind for the Measurability task force. 
I think another more relevant task force should work on how can tools help check/monitor compliance to the WSG, taking inspiration from what exists for WCAG : which guidelines could be checked automatically (Lighthouse and such) ? How do we help check the other guidelines (though support for manual testing, for example => Sustainability Advisor) ?

**Comment by hidde** [comment]
> What kinds of tools are there and what kind of information is it that they can collect? What can we learn from the tools and data which can be gathered in the accessibility world? 

Another type of tooling that I didn't see in this thread and that could be useful: someting to create a report that can be used in a sustainability statement of sorts. 

I'm thinking something like that [WCAG-EM Report Tool](https://www.w3.org/WAI/eval/report-tool/) or [ATAG Report Tool](https://www.w3.org/WAI/atag/report-tool/), where you can state a status for individual success criteria and add notes for specifics of conformance/non-conformance.

**Comment by mgifford** [comment]
Having something like that built in that could possibly feed into a Carbon.txt file would be useful. 

I like how we extended your work to allow for both a machine-readable download and a HTML version for us humans.
https://acreditor.section508.gov/
